### PR TITLE
fix(snap): Refactor to avoid conflicts with readonly config provider directory

### PR DESCRIPTION
**Pull request by MonicaisHer**
This PR makes changes to incorporate the following fixes:
- https://github.com/canonical/edgex-snap-hooks/pull/83
- https://github.com/canonical/edgex-snap-hooks/pull/82

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-service-configurable/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

For testing instructions, please refer to https://github.com/edgexfoundry/device-mqtt-go/pull/535

### CODE REVIEW

File: snap/local/bin/source-env-file.sh [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
-SERVICE="app-service-configurable"
-ENV_FILE="$SNAP_DATA/config/res/$SERVICE.env"
+# convert cmdline to string array
+ARGV=($@)
+
+# grab binary path
+BINPATH="${ARGV[0]}"
+
+# binary name == service name/key
+SERVICE=$(basename "$BINPATH")
```
Comment (farshidtz) [inline snap/local/bin/source-env-file.sh:10]: Has this been tested?
I think `basename $BINPATH` is going to give you the name of this script, not the wanted name of the app. If so, please revert this change.

File: snap/local/bin/source-env-file.sh [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+ENV_FILE="$SNAP_DATA/config/$SERVICE/overrides.env"
```
Comment (farshidtz) [inline snap/local/bin/source-env-file.sh:11]: The given path is incorrect. See https://github.com/canonical/edgex-snap-hooks/blob/544c6823a1b007cf18e22cfa85d1ba12b0ffa2ba/options/env.go#L81

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/local/bin/source-env-file.sh (hunk 1 above)
Comment (MonicaisHer) [reply snap/local/bin/source-env-file.sh]: I have reversed it. `basename $BINPATH` should equal `app-service-configurable` as expected, not `service-wrapper.sh`.

**Review by farshidtz — APPROVED** [review]
